src/devices/bus/mononcol/slot.cpp

```diff
@@ -0,0 +1,185 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli
+
+#include "emu.h"
+#include "slot.h"
+
+#define VERBOSE 0
+#include "logmacro.h"
+
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(MONONCOL_SOCKET, mononcol_socket_device, "mononcol_socket", "Monon Color Socket")
+DEFINE_DEVICE_TYPE(MONONCOL_CARTSLOT, mononcol_cartslot_device, "mononcol_cartslot", "Monon Color Cartridge Slot")
+
+
+//**************************************************************************
+//  device_mononcol_cart_interface
+//**************************************************************************
+
+device_mononcol_cart_interface::device_mononcol_cart_interface(machine_config const &mconfig, device_t &device) :
+	device_interface(device, "genslot"),
+	m_region(*this, DEVICE_SELF),
+	m_rom(nullptr),
+	m_rom_size(0)
+{
+}
+
+device_mononcol_cart_interface::~device_mononcol_cart_interface()
+{
+}
+
+void device_mononcol_cart_interface::rom_alloc(u32 size, int width, endianness_t endian, char const *tag)
+{
+	if (m_rom)
+	{
+		throw emu_fatalerror(
+				"%s: Request to allocate ROM when already allocated (allocated size %u, requested size %u)\n",
+				device().tag(),
+				m_rom_size,
+				size);
+	}
+
+	std::string fulltag(tag);
+	fulltag.append(MONONCOL_ROM_REGION_TAG);
+	device().logerror("Allocating %u byte ROM region with tag '%s' (width %d)\n", size, fulltag, width);
+	m_rom = device().machine().memory().region_alloc(fulltag.c_str(), size, width, endian)->base();
```

Comment:
`region_alloc` takes a string by value for the tag – there’s no need to use `.c_str()` here.

Reply:
likewise, you should probably fix this in generic/slot.cpp, as that is the goto example of how to do this stuff (being the most generic / basic case)

Reply:
> likewise, you should probably fix this in generic/slot.cpp, as that is the goto example of how to do this stuff (being the most generic / basic case)

As I said before, it’s a bad example because it’s over-complicated by trying to be too many things to too many people.  It’s also overly convoluted.  I started trying to fix it a while ago, but it’s too tightly tangled up with too many things for that to be a simple task.